ios/Classes/PurchasesFlutterPlugin.m

```diff
@@ -526,11 +560,23 @@ - (void)rejectWithResult:(FlutterResult)result error:(RCErrorContainer *)errorCo
     };
 }
 
+- (void (^)(RCErrorContainer *))getBeginRefundResponseCompletionBlock:(FlutterResult)result {
+    return ^(RCErrorContainer * _Nullable error) {
+        if (error == nil) {
+            result(@0);
+        } else if ([error.info[@"userCancelled"] isEqual: @YES]) {
```

Comment:
So, I originally wanted to just expose the error in the flutter side, but after testing it out, accessing this property on the error in the flutter side was very verbose and difficult... Due to that, I decided to just mimick the API in iOS which gives an enum with 3 possible values. 

The part I don't like of this is that now we have to process the error from PHC to map to the enum, instead of just returning an integer from PHC and avoiding this extra processing in all hybrids... So I was thinking about changing PHC but that would technically be a breaking change there... So in the end I decided to map the existing response. Lmk what you think!

Reply:
This makes sense. PHC is generic but here you're making the best API for this particular platform.